File: JAVA/fizzbuzz/Solution/src/main/java/com/example/fizzbuzz/FizzBuzz.java
package com.example.fizzbuzz;

public class FizzBuzz {
    public String transform(int number) {
        if (isMulpipleOf3(number) && isMultipleOf5(number))
            return "FizzBuzz";
        if(isMulpipleOf3(number))
            return "Fizz";
        if (isMultipleOf5(number))
            return "Buzz";
        return String.valueOf(number);
    }

    private static boolean isMultipleOf5(int number) {
        return number % 5 == 0;
    }

    private static boolean isMulpipleOf3(int number) {
        return number % 3 == 0;
    }
}

File: Typescript/Fizzbuzz/Solution/src/fizzbuzz.ts
export class Fizzbuzz {
    transform(number: number): string {
        if (this.isMultipleOfThree(number) && this.isMultipleOfFive(number))
            return "FizzBuzz";
        if (this.isMultipleOfThree(number))
            return "Fizz";
        if (this.isMultipleOfFive(number))
            return "Buzz";
        return number.toString();
    }

    private isMultipleOfFive(number: number) {
        return number % 5 === 0;
    }

    private isMultipleOfThree(number: number) {
        return number % 3 === 0;
    }
}
